File: day3.cpp
class Solution
{
public:
    vector<bool> checkArithmeticSubarrays(vector<int> &nums, vector<int> &l, vector<int> &r)
    {

        int n = l.size();

        vector<bool> result;

        for (int i = 0; i < n; i++)
        {
            int lower = l[i];
            int upper = r[i];
            int size = upper - lower + 1;
            int temp[size];
            int count = 0;
            for (int j = lower; j <= upper; j++)
            {
                temp[count] = nums[j];
                count++;
            }
            sort(temp, temp + size);
            int diff = temp[1] - temp[0];
            bool found = 0;
            for (int j = 2; j < size; j++)
            {
                if (temp[j] - temp[j - 1] != diff)
                {
                    result.push_back(false);
                    found = 1;
                    break;
                }
                diff = temp[j] - temp[j - 1];
            }
            if (found)
                continue;
            result.push_back(true);
        }

        return result;
    }
};

File: Day3.py
# Day 3
# Question : https://www.codechef.com/START3A/problems/HIRETEST
# 26/05/2022
def ap_checker(Lisht):
    diff=Lisht[1]-Lisht[0] #finding C.D as constraints say number_elemnts in num>=2;
    for i in range(1,len(Lisht)-1):
        if Lisht[i+1]-Lisht[i]!=diff:
            return 0
    return 1 #func checks for Ap as name states.
class Solution(object):
    def checkArithmeticSubarrays(self,nums, l, r): #l,r have m integers; nums has n integers
        bool_list=[]
        for i,j in zip(l,r):
            subArray=nums[i:j+1]
            subArray=sorted(subArray)
            bool_list.append(ap_checker(subArray))
        return bool_list